### DISCUSSION TITLE
ble advertising example

### DISCUSSION THREAD

**Discussion by uraich** [micropython/micropython#12701] [ESP32]
I am currently trying to learn about BlueThooth on the ESP32 and I tried the ble_advertising.py example.
When running it, I get: ValueError: advertising payload too large
The payload length is 38 with a maximum of 32 allowed. The problem could easily be solved by modifying the name parameter of advertising_payload from "micropython" to "mpy".

**Comment by jimmo** [comment]
@uraich That's right (but the limit is 31 bytes, not 32). You can get a further 31 bytes using the scan response payload.

> The problem could easily be solved by modifying the name parameter of advertising_payload from "micropython" to "mpy".

I'm not sure what you mean by this? `ble_advertising.py` doesn't set the name by default.

**Reply by uraich** [reply]
What I mean is this:
In the function demo of ble_advertising.py I replaced "micropython" by "mpy".
![Screenshot from 2023-10-17 14-08-03](https://github.com/micropython/micropython/assets/1673533/54eae52e-442c-4f3b-b615-bed9ec89e916)

**Comment by FilMarini** [comment]
@jimmo Hi, I'm having the same issue, where for my application I need to set a specific name for advertising but it results too long. Changing it is, unfortunately, not an option.
How can I get it further than 31 bytes? Is there an example of using the scan response payload?
Thanks
